e2e/rook-test-framework/enums/rook_platform_types.go

```diff
@@ -0,0 +1,41 @@
+package enums
+
+import (
+	"errors"
+	"strings"
+)
+
+type RookPlatformType int
+
+const (
+	Kubernetes RookPlatformType = iota + 1
+	BareMetal
+	StandAlone
+	None
+)
+
+var platforms = [...]string{
+	"Kubernetes",
+	"BareMetal",
+	"StandAlone",
+	"None",
+}
+
+func (platform RookPlatformType) String() string {
+	return platforms[platform-1]
+}
+
+func GetRookPlatFormTypeFromString(name string) (RookPlatformType, error) {
+	switch {
+	case strings.EqualFold(name, Kubernetes.String()):
```

Comment:
This method returns an int that is of type RookPlatformType, which is basically a strongly typed int/enum.  The method is looking up to see if the string name is a defined type.